lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -1219,7 +1246,7 @@ SILCombiner::visitInjectEnumAddrInst(InjectEnumAddrInst *IEAI) {
   //  store %1 to %nopayload_addr
   //
   auto *AI = dyn_cast_or_null<ApplyInst>(getSingleNonDebugUser(DataAddrInst));
-  if (!AI)
+  if (!AI || !DI->properlyDominates(AI, IEAI))
```

Comment:
Should the dominance check be limited to empty types ? For non empty types, one path could have the initializing apply, and another path may end in an unreachable ?

Reply:
> With the check we may not be optimizing the case where one path could have the initializing apply, and another uninitialized path may end in an unreachable.
It seems like, even if the address is uninitialized on some `unreachable`-terminated path, every path through `IEAI` would still go through `AI``*` (`AI` would dominate `IEAI`).  But...
> For non empty types we do not have uninitialized paths and won't need this check?
Great point!  This check is wasteful for non-empty types.  Added a check that the type is empty before checking dominance.  And added some test cases with dead end blocks after the `init_enum_data_addr`.

`*` Or some other initialization in which case there's another bailout since the apply case only handles the case where the apply is the single use.